### ISSUE TITLE
linter: max-lines-per-function defaults to 0 when enabling pedantic category

### ISSUE THREAD

**Issue by ChooKing** [oxc-project/oxc#11907]
When using this setting in .oxlintrc.json
```
"categories": {
    "pedantic": "warn"
  }
```
And this minimal code:
```ts
export class Foo{
    constructor(){
        console.log("Hello World!");
    }
}
```
Expected result: no warning
Observed: " eslint(max-lines-per-function): The constructor `constructor` has too many lines (3). Maximum allowed is 0."

Note: Removing the category rule and using this config instead produces no warning:
```
"rules": {
    "max-lines-per-function": "warn"
}
```
Additional notes:
The ESLint default is 50 and not 0.  Using oxc version 1.3.0, but also observed this bug in earlier versions.
